src/video/aml.c

```diff
@@ -39,11 +39,15 @@
 
 #define SYNC_OUTSIDE 0x02
 #define UCODE_IP_ONLY_PARAM 0x08
+#define DECODER_BUFFER_SIZE 512*1024
+#define MAX_WRITE_ATTEMPTS 3
+#define EAGAIN_SLEEP_TIME 15 * 1000
```

Comment:
15 ms is quite a while to sleep. It basically guarantees we will drop a frame.

Maybe we should allow more attempts with shorter waiting periods like 1 or 2 ms.